packages/storybook-a11y-test/src/index.ts

```diff
@@ -115,22 +123,43 @@ const testPage = (iframePath, browser, timeout, disableAnimation) => {
             }`,
         });
       }
-      const result = await page.evaluate(() =>
-        window.axe.run(document.getElementById('root'), {}),
-      );
+      const result: axeCore.AxeResults = await page.evaluate((id) => {
+        const story = window.__STORYBOOK_STORY_STORE__.fromId(id)!;
+
+        // optional selector which element to inspect
+        const axeElement: axeCore.ElementContext | undefined =
+          story.parameters.a11y && story.parameters.a11y.element;
+
+        // axe-core configurationOptions (https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#parameters-1)
+        const axeConfig: axeCore.Spec =
+          (story.parameters.a11y && story.parameters.a11y.config) || {};
+
+        // axe-core optionsParameter (https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#options-parameter)
+        const axeOptions: axeCore.RunOptions =
+          (story.parameters.a11y && story.parameters.a11y.options) || {};
+
+        window.axe.configure(axeConfig);
+
+        return window.axe
+          .run(axeElement || document.getElementById('root')!, axeOptions)
+          .then((results) => {
+            return results;
+          })
```

Comment:
This then block seems to return exactly what was passed to it so it doesn't seem to do anything and thus can be removed


```suggestion
```